src/service/standard-actions-impl.js

```diff
@@ -198,6 +238,30 @@ export class StandardActions {
     );
   }
 
+  /**
+   * Handles the `toggleTheme` action.
+   *
+   * This action sets the `amp-dark-mode` class on the body element and stores the the preference for dark mode in localstorage.
+   */
+  handleToggleTheme_() {
+    this.ampdoc.waitForBodyOpen().then((body) => {
+      try {
+        const darkMode = this.prefersDarkMode_();
+        if (darkMode) {
+          body.classList.remove('amp-dark-mode');
+          this.ampdoc.win.localStorage.setItem('amp-dark-mode', 'no');
+        } else {
+          body.classList.add('amp-dark-mode');
+          this.ampdoc.win.localStorage.setItem('amp-dark-mode', 'yes');
+        }
+      } catch (e) {
+        // LocalStorage may not be accessible
+        // toggle should work even if localStorage is not available.
+        body.classList.toggle('amp-dark-mode');
```

Comment:
We will have already toggled the darkmode, since only the localStorage could fail. I think this will just reset us to the previous state?

Reply:
Yes, correct thanks for catching that I will remove it.